src/deployment/deployment.go

```diff
@@ -405,3 +406,38 @@ func CreateDeployment(sp *serviceprovider.Container, deploy *entity.Deployment)
 func DeleteDeployment(sp *serviceprovider.Container, deploy *entity.Deployment) error {
 	return sp.KubeCtl.DeleteDeployment(deploy.Name, deploy.Namespace)
 }
+
+// CreateAutoscaler will create a autoscaler
+func CreateAutoscaler(sp *serviceprovider.Container, autoscalerInfo entity.AutoscalerInfo) error {
+	autoscaler := v2beta1.HorizontalPodAutoscaler{
+		ObjectMeta: metav1.ObjectMeta{
+			// use deployment name to name autoscaler's name
+			Name:      autoscalerInfo.ScaleTargetRefName,
+			Namespace: autoscalerInfo.Namespace,
+		},
+		Spec: v2beta1.HorizontalPodAutoscalerSpec{
+			ScaleTargetRef: v2beta1.CrossVersionObjectReference{
+				Kind: "Deployment",
+				Name: autoscalerInfo.ScaleTargetRefName,
+			},
+			MinReplicas: &autoscalerInfo.MinReplicas,
+			MaxReplicas: autoscalerInfo.MaxReplicas,
+			Metrics: []v2beta1.MetricSpec{
+				{
+					Type: v2beta1.ResourceMetricSourceType,
+					Resource: &v2beta1.ResourceMetricSource{
+						Name:                     autoscalerInfo.ResourceName,
+						TargetAverageUtilization: &autoscalerInfo.TargetAverageUtilization,
+					},
+				},
+			},
+		},
+	}
+	_, err := sp.KubeCtl.CreateAutoscaler(&autoscaler, autoscalerInfo.Namespace)
+	return err
+}
+
+// DeleteAutoscaler will delete a autoscaler
+func DeleteAutoscaler(sp *serviceprovider.Container, autoscalerInfo entity.AutoscalerInfo) error {
```

Comment:
Will autoscaler delete itself while the corresponding deployment is been removed ?